### PR TITLE
[#286] Fix: Terraform plan can't be applied because of VPC endpoint misconfiguration

### LINKED ISSUE #286: Fix: Terraform plan can't be applied because of VPC endpoint misconfiguration
## Issue

When we are generating new project and trying to apply it we have an error from AWS API:

> Error: creating EC2 VPC Endpoint (com.amazonaws.ap-southeast-1.logs): InvalidParameter: Endpoint type (Gateway) does not match available service types ([Interface]). status code: 400, request id: 1b6ee40d-2c83-4f03-b98e-3b8d05240989

## Expected

Generated code should work without issues.

### PR DESCRIPTION
**Pull request by Nihisil**
- Close #286

## What happened 👀

- Revert changes from https://github.com/nimblehq/infrastructure-templates/pull/225

## Insight 📝

VPC endpoints are a good way to optimize because they lower traffic costs. However, they also come with hourly charges, and in cases where there is not much traffic, using a VPC endpoint might end up being more expensive than just paying for the traffic. I believe the decision to add a VPC endpoint should be made on a per-project basis and should not be included in the template by default. For most projects, it makes sense to have VPC endpoints for development with regular deployments and debug logs, but for staging and production, it might be better not to use VPC endpoints.

## Proof Of Work 📹

Before the fix we weren't able to apply TF plan generated from our code:

image

After the fix TF plan can be applied without error ✅

### CODE REVIEW

**Review by malparty — APPROVED** [review]
Thanks for your insights. This makes sense indeed 🙇

**Review by byhbt — APPROVED** [review]

**Review by hoangmirs — APPROVED** [review]

File: .github/wiki/assets/images/architecture/diagram-complete.svg
Comment (sanG-github) [inline .github/wiki/assets/images/architecture/diagram-complete.svg:1]: We don't have the VPC Endpoint for S3, however the change represents it. Please update the latest version 🙏🏻

**Review by sanG-github — COMMENTED** [review]
The changes look good to me, here is some insights to when should we consider using the VPC Endpoint instead of NAT Gateway.

**TL;DR:** Use VPCE for a service when the total of traffic is `higher than 160GB per month` to save cost from charging of hourly billing for VPCE.

Here is the [NAT Pricing](https://aws.amazon.com/vpc/pricing/) (based on `us-east-1` region):

> NAT Gateway Hourly Charge: NAT Gateway is charged on an hourly basis. For this region, the rate is $0.045 per hour.
> NAT Gateway Data Processing Charge: 1 GB data went through the NAT gateway. The Data Processing charge will result in a charge of $0.045.

And here is the [VPCE Pricing](https://aws.amazon.com/privatelink/pricing/) (based on `us-east-1` region):

> Pricing per VPC endpoint per AZ ($/hour) $0.01
> Data Processing Charge: First 1 PB, $0.01/GB

Therefore, as 0.01$/hour x 720 = 7.2$ for 1 VPC endpoint per month ~ which is equivalent to `160GB` data charged by NAT (0.045$/GB x 160GB = 7.2$), as we already have the NAT Gateway in our system.

File: .github/wiki/assets/images/architecture/diagram-complete.svg
Comment (hoangmirs) [reply .github/wiki/assets/images/architecture/diagram-complete.svg:1]: Revert to the previous version in https://github.com/nimblehq/infrastructure-templates/pull/309/commits/35209b94f38985bf825aaae61f0bd3c578f96a78

Comment (sanG-github) [reply .github/wiki/assets/images/architecture/diagram-complete.svg:1]: Thank you! 🙏🏻

**Review by sanG-github — APPROVED** [review]

**Review by tung-nimblehq — APPROVED** [review]
